### PR TITLE
Use upcoming compile api for fea-rs

### PR DESCRIPTION
**Pull request by rsheeter**
Build against https://github.com/cmyr/fea-rs/pull/119.

Ran into some issues:

1. Had to copy the feature content for in memory use
1. Using a closure for the resolve sounded lovely until I tried to actually do it, then I angered the compiler and ultimately made a struct
1. `cargo test -p fontc fea` now fails, apparently absolute paths aren't welcome?
   * `/blah-blah-blah/fontmake-rs/resources/testdata/Static-Regular.ufo/features.fea` triggers `thread 'tests::compile_fea' panicked at 'assertion failed: path.is_relative()', ...blah blah.../fea-rs/src/util/paths.rs:15:9

Update: ^ fixed by latest updates to the fea-rs branch.

### CODE REVIEW

File: fontbe/src/features.rs [hunk 1]
```diff
@@ -18,114 +20,59 @@ use crate::{
...
+    content_path: OsString,
+    content: String,
+}
+
+impl SourceResolver for CursedClosures {
+    fn get_contents(&self, path: &OsStr) -> Result<String, SourceLoadError> {
+        if path == &*self.content_path {
+            return Ok(self.content.clone());
         }
+        // I don't think I can actually construct this given private fields and pub(crate) new
```
Comment (cmyr) [inline fontbe/src/features.rs:45]: mm good point, will look into this.

File: fontbe/src/features.rs [hunk 2]
```diff
@@ -18,114 +20,59 @@ use crate::{
...
+            let resolver = CursedClosures {
+                content_path: root_path.clone(),
+                content: fea_content.clone(),
+            };
+            compiler = compiler.with_resolver(resolver);
+        }
+        match compiler.compile() {
+            Ok(compilation) => compilation
+                .assemble(&glyph_order, Default::default())
+                .map_err(|e| Error::FeaError(format!("{:?} assembling {:?}", e, root_path))),
```
Comment (cmyr) [inline fontbe/src/features.rs:70]: I would make your `FeaError` variant just wrap the error produced by `fea-rs`.

**Review by cmyr — COMMENTED** [review]
Some notes inline, I agree this doesn't feel like it's quite there yet on  the `fea-rs` side, although I'm not exactly sure what the answer is.

File: fontbe/src/features.rs (hunk 1 above)
Comment (rsheeter) [reply fontbe/src/features.rs:45]: Fixed in fea-rs update

File: fontbe/src/features.rs [hunk 3]
```diff
@@ -18,114 +20,59 @@ use crate::{
...
-        if parse.is_err() {
-            write_debug_fea(context, parse.is_err(), "fea parse failed", fea_content);
+        eprintln!("ROOT: {:?}", root_path);
+        let mut compiler = Compiler::new(root_path.clone(), &glyph_order);
+        if let Features::Memory(fea_content) = features {
+            let resolver = CursedClosures {
+                content_path: root_path.clone(),
+                content: fea_content.clone(),
+            };
+            compiler = compiler.with_resolver(resolver);
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply fontbe/src/features.rs:65]: Features::File uses your default impl, Features::Memory provides it's own so it's not quite true the whole Features type is the resolver.

[the remark this answers is not in the document]
Comment (rsheeter) [reply fontbe/src/features.rs]: For now we fail resolution of anything other than the root item. https://github.com/googlefonts/fontmake-rs/issues/109 covers actual resolution.

**Pull request merged**